index.js

```diff
@@ -1,8 +1,99 @@
-const net = require('net');
+'use strict';
+
+const fs = require('fs');
+const http = require('http');
+const path = require('path');
+
 const server = require('./src/server');
 
+const tempDir = path.join(__dirname, 'tmp')
+const statusFilePath = path.join(tempDir, 'status.txt')
+
 module.exports = {
-    startServer: (port) => {
-        return server.start(port || 0);
+    startServer: () => {
+        let currentServerPort;
+        return getState()
+            .then((port) => {
+                return checkServerHelth(port)
+                    .then(serverHealth => {
+                        currentServerPort = port
+                        if (!serverHealth) {
+                            return server.start(port)
```

Comment:
Why try and start the server on the same port when it has died out?
If you do not try and start it on the same port but proceed to call `initServerWithFreePort` directly, this code would 100% match the one in the second argument ot the first `then` statement below and you could extract the functionality in a separate function